utils.js

```diff
@@ -54,13 +48,11 @@ exports.normalizeScopeSets = function(scopesets) {
  * Also expressed as ('a' and 'b') or 'c'.
  */
 exports.scopeMatch = function(scopePatterns, scopesets) {
-  var scopesets = exports.normalizeScopeSets(scopesets);
+  exports.validateScopeSets(scopesets);
   if (typeof(scopePatterns) == 'string') {
     scopePatterns = [scopePatterns];
   }
```

Comment:
it takes a few extra iterations of the loops.. but it's probably worth to validate it on every turn for all values passed in...
So maybe we should do:
```js
exports.scopeMatch = function(scopePatterns, scopesets) {
  exports.validateScopeSets(scopesets);
  assert(scopePatterns instanceof Array && scopePatterns.every(function(scope) {
    return typeof(scope) === 'string';
  }), "scopes must be an array of strings");
  return scopesets.some(function(scopeset) {
    return scopeset.every(function(scope) {
      return scopePatterns.some(function(pattern) {
        if (scope === pattern) {
          return true;
        }
        if (/\*$/.test(pattern)) {
          return scope.indexOf(pattern.slice(0, -1)) === 0;
        }
        return false;
      });
    });
  });
```
So we also validate the sanity of `scopePatterns` it's just as important.. And should always come from a source that provides it as a list...